sdk/identity/identity/src/credentials/managedIdentityCredential/index.ts

```diff
@@ -297,6 +297,19 @@ export class ManagedIdentityCredential implements TokenCredential {
         throw err;
       }
 
+      // This is a special case for Docker Desktop which responds with a 403 with a message that contains "A socket operation was attempted to an unreachable network"
+      // rather than just timing out, as expected.
+      if(err.statusCode === 403){
```

Comment:
I'd probably move this to after the section further down where you check if `err.statusCode == 400` to keep all the statusCode checks together.